lib/ansible/modules/network/nxos/nxos_rpm.py

```diff
@@ -0,0 +1,269 @@
+            else:
+                return
+        else:
+            return
+
+
+def validate_operation(module, show_cmd, cfg_cmd, pkg, pkg_not_present):
+    iteration = 0
+    while iteration < 10:
+        body = execute_show_command(show_cmd, module)
+        if pkg_not_present:
+            if pkg not in body:
+                return
+        else:
+            if pkg in body:
+                return
+        time.sleep(2)
+        iteration += 1
+
+    err = 'Operation "{0}" Failed'.format(cfg_cmd)
+    module.fail_json(msg=err)
+
+
+def add_operation(module, show_cmd, file_system, pkg, fixed_pkg):
+    cmd = 'install add {0}:{1}'.format(file_system, pkg)
+    config_cmd_operation(module, cmd)
+    validate_operation(module, show_cmd, cmd, fixed_pkg, False)
+    return cmd
+
+
+def activate_operation(module, show_cmd, fixed_pkg):
+    cmd = 'install activate {0} forced'.format(fixed_pkg)
+    config_cmd_operation(module, cmd)
+    validate_operation(module, show_cmd, cmd, fixed_pkg, False)
+    return cmd
+
+
+def commit_operation(module, show_cmd, fixed_pkg):
+    cmd = 'install commit {0}'.format(fixed_pkg)
+    config_cmd_operation(module, cmd)
+    validate_operation(module, show_cmd, cmd, fixed_pkg, False)
+    return cmd
+
+
+def deactivate_operation(module, show_cmd, fixed_pkg, flag):
+    cmd = 'install deactivate {0} forced'.format(fixed_pkg)
+    config_cmd_operation(module, cmd)
+    validate_operation(module, show_cmd, cmd, fixed_pkg, flag)
+    return cmd
+
+
+def remove_operation(module, show_cmd, fixed_pkg):
+    cmd = 'install remove {0} forced'.format(fixed_pkg)
+    config_cmd_operation(module, cmd)
+    validate_operation(module, show_cmd, cmd, fixed_pkg, True)
+    return cmd
+
+
+def install_remove_rpm(module, pkg, file_system, state):
+    commands = []
+
+    splitted_pkg = pkg.split('.')
+    fixed_pkg = '.'.join(splitted_pkg[0:-1])
+
+    show_inactive = 'show install inactive'
+    show_active = 'show install active'
+    show_commit = 'show install committed'
+    show_patches = 'show install patches'
+
+    if state == 'present':
+        inactive_body = execute_show_command(show_inactive, module)
+        active_body = execute_show_command(show_active, module)
+
+        if fixed_pkg not in inactive_body and fixed_pkg not in active_body:
+            commands.append(add_operation(module, show_inactive, file_system, pkg, fixed_pkg))
+
+        if fixed_pkg not in active_body:
+            commands.append(activate_operation(module, show_active, fixed_pkg))
+
+        commit_body = execute_show_command(show_commit, module)
+        if fixed_pkg not in commit_body:
+            patch_body = execute_show_command(show_patches, module)
+            if fixed_pkg in patch_body:
+                # this is an smu
+                commands.append(commit_operation(module, show_active, fixed_pkg))
+            else:
+                err = 'Operation "install activate {0} forced" Failed'.format(fixed_pkg)
```

Comment:
Do you ever git into this else block or is this just defensive coding?  U would expect `validate_operation` to catch most if not all of the errors right?